workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/PropertyInfoAdapter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Reflection;
+
+namespace Improbable.Gdk.Core
+{
+    public class PropertyInfoAdapter : IMemberAdapter
+    {
+        private readonly PropertyInfo member;
+
+        public MemberInfo Member
+        {
+            get { return member; }
+        }
+
+        public PropertyInfoAdapter(PropertyInfo member)
+        {
+            this.member = member;
+        }
+
+        public void SetValue(object obj, object value)
```

Comment:
These can be specialised. E.g. we know that obj is always going to be a monobehaviour

Reply:
I actually kinda like the general nature of the current implementation. An argument can be made that this class could in theory be used for arbitrary classes. We would probably also need to call the class `MonoBehaviourPropertyInfoAdapter` otherwise which feels overkill. Just my two cents.

Reply:
Also right now I'm not sure if the readers are actually going to have a supertype and if so what, so keeping them general helps not force that decision before other PRs are in

Reply:
Let's not over-design something on the chance that it *may* be used at some point. Solve exactly the problem you intend to solve now/

Reply:
If we start doing lots of arbitrary reflection then arbitrary utils classes makes sense. While it's just one specific class it doesn't though.